src/integration_tests/mission_takeoff_land.cpp

```diff
@@ -0,0 +1,171 @@
+
+    Mission::MissionItem new_item{};
+    Mission::MissionPlan mission_plan{};
+
+    new_item.latitude_deg = 47.398170327054473;
+    new_item.longitude_deg = 8.5456490218639658;
+    new_item.relative_altitude_m = mission_altitude_m;
+    new_item.vehicle_action = Mission::MissionItem::VehicleAction::Takeoff;
+    mission_plan.mission_items.push_back(new_item);
+
+    new_item.latitude_deg = 47.398370327054473;
+    new_item.vehicle_action = Mission::MissionItem::VehicleAction::Land;
+    mission_plan.mission_items.push_back(new_item);
+
+    new_item.latitude_deg = 47.398570327054473;
+    new_item.vehicle_action = Mission::MissionItem::VehicleAction::Takeoff;
+    mission_plan.mission_items.push_back(new_item);
+
+    new_item.latitude_deg = 47.398770327054473;
+    new_item.vehicle_action = Mission::MissionItem::VehicleAction::Land;
+    mission_plan.mission_items.push_back(new_item);
+
+    {
+        LogInfo() << "Uploading mission...";
+        auto prom = std::make_shared<std::promise<void>>();
+        auto future_result = prom->get_future();
+        mission->upload_mission_async(mission_plan, [prom](Mission::Result result) {
+            ASSERT_EQ(result, Mission::Result::Success);
+            prom->set_value();
+            LogInfo() << "Mission uploaded.";
+        });
+
+        auto status = future_result.wait_for(std::chrono::seconds(2));
+        ASSERT_EQ(status, std::future_status::ready);
+        future_result.get();
+    }
+
+    pc.set_min_altitude(0.0f);
+    pc.set_next_reach_altitude(mission_altitude_m);
+    pc.set_max_altitude(mission_altitude_m);
+
+    LogInfo() << "Arming...";
+    const Action::Result arm_result = action->arm();
+    ASSERT_EQ(arm_result, Action::Result::Success);
+    LogInfo() << "Armed.";
+
+    // Before starting the mission, we want to be sure to subscribe to the mission progress.
+    mission->subscribe_mission_progress([](Mission::MissionProgress progress) {
+        LogInfo() << "Mission status update: " << progress.current << " / " << progress.total;
+    });
+
+    {
+        LogInfo() << "Starting mission.";
+        auto prom = std::make_shared<std::promise<void>>();
+        auto future_result = prom->get_future();
+        mission->start_mission_async([prom](Mission::Result result) {
+            ASSERT_EQ(result, Mission::Result::Success);
+            prom->set_value();
+            LogInfo() << "Started mission.";
+        });
+
+        auto status = future_result.wait_for(std::chrono::seconds(2));
+        ASSERT_EQ(status, std::future_status::ready);
+        future_result.get();
+    }
+
+    while (!mission->is_mission_finished().second) {
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+
+    while (telemetry->armed()) {
+        // Wait until we're done.
+        std::this_thread::sleep_for(std::chrono::seconds(1));
+    }
+    for (int i = 0; i < 8; i++) {
+        ASSERT_EQ(landed_states_template, landed_states);
+    }
```

Comment:
Either you want to do `landed_states_template.at(i)` and `landed_states.at(i)`, or remove the for loop (not sure if that works?), but something is wrong here :see_no_evil:

Reply:
I just wanted to make sure, that those two vector are same. So I did check 8 times. 🙈🙈 My mistake. already fixed.